metals/src/main/scala/scala/meta/internal/metals/BatchedFunction.scala

```diff
@@ -57,9 +57,11 @@ final class BatchedFunction[A, B](
     unlock()
   }
 
-  def cancelCurrentRequest(): Unit = {
-    current.get().cancelable.cancel()
```

Comment:
[note]
So, the problem was, when we cancel the compilation, it cancels the currently running compilation but it doesn't for the queued compilation.

Reply:
[question]
Can't we cancel the "ongoingCompilation" by `current.get().cancelable.cancel()`?
I'm wondering maybe canceling this cascadingly cancel the ongoing compilation too 🤔  https://github.com/scalameta/metals/blob/de34e01ef61b28c36250b0f6b2c3cdd4254c6d95/metals/src/main/scala/scala/meta/internal/metals/Compilations.scala#L243

Reply:
> So, the problem was, when we cancel the compilation, it cancels the currently running compilation but it doesn't for the queued compilation.
> Am I right?

That's one of the issues, the other one was that cancellation was not propagated to the final future.

> Can't we cancel the "ongoingCompilation" by current.get().cancelable.cancel()?
> I'm wondering maybe canceling this cascadingly cancel the ongoing compilation too thinking

I couldn't make that work and it's super easy to break and not have the cancellation properly propagated.